### PR TITLE
Report JIT allocated bytes and current RSS in tests

### PR DESCRIPTION
**Pull request by staticfloat**
This is primarily to help debug our memory problems on i686-linux-gnu

### CODE REVIEW

**Comment by staticfloat** [comment]
As an aside, my worry that we're allocating hundreds of MB's of generated code do not seem founded; it seems we're generally under 30MB of generated code, which is reasonable.

I also wonder if it makes more sense to recycle worker processes after residual live set is greater than a threshold, rather than when max RSS is greater than a threshold.

**Review by KristofferC — APPROVED** [review]

**Comment by staticfloat** [comment]
I'm now dumping out `/proc/self/status` and parsing `VmSize` to see if I can catch likely address space exhaustion

**Comment by staticfloat** [comment]
Looks likely!  The tests happened to pass on i686-linux-gnu this last time, but we can see that the `Pkg` worker process [had a perilously high `VmSize`](https://buildkite.com/julialang/julia-master/builds/12581#01812b3e-0aa6-47fc-b23c-9018a1e78f0a/496-1171) of `3169.1 MB`.  I'm running a new test now to break the `VmSize` down into its substituent pieces on Linux.

**Comment by staticfloat** [comment]
@vtjnash I'd like to trigger the equivalent of a `cat /proc/$(getpid())/stat` upon process exit for the workers.  It could happen in a segfault handler, or through some kind of exit hook, I don't really care.  What do you recommend?  Should I run the test workers inside of `gdb` and use a gdb batch script or something?

File: base/sysinfo.jl [hunk 1]
```diff
@@ -292,6 +292,13 @@ See also:
 """
 maxrss() = ccall(:jl_maxrss, Csize_t, ())
 
+"""
+    Sys.rss()
+
+Get the current resident set size utilized in bytes.
+"""
+rss() = ccall(:jl_rss, Csize_t, ())
```
Comment (vtjnash) [inline base/sysinfo.jl:300]: seems like this could use `uv_resident_set_memory`  directly

**Comment by vtjnash** [comment]
I think we already print some GC info if we die, but when faced with SIGKILL, there is no chance to run that code. You would need a separate watcher process to be an intermediate (think `/usr/bin/time` like). I have done that before with `rr` by making a new folder next to bin/julia with the content and setting JULIA_BINDIR to that folder instead.
```
#!/bin/sh
exec rr record --nested=detach "$(dirname $(dirname $0))/bin/julia" "$@"
```
A bit trickier if the extra output will confuse the parent though. I think this might work, but I haven't tested at all:
```
#!/bin/sh
exec /usr/bin/time sh -s -- "$@" 3<<&0 4>>&1 5>>&2 2<<time.out 1<<&2 0<<EOF
exec "$(dirname $(dirname $0))/bin/julia" "\$@" 0<<&3 1>>&4 2>>&5 3>&- 4>&- 5>&- 
EOF
```

Oh, or probably just this would do the same but simpler:
```
#!/bin/sh
exec /usr/bin/time -o time.out "$(dirname $(dirname $0))/bin/julia" "$@"
```

**Comment by staticfloat** [comment]
Now that https://github.com/JuliaCI/julia-buildkite/commit/3ec7e645526a4b60698f38a4d0a4c828b93875fb is merged, I think we will start getting coredumps when Linux i686 OOMs.  From those, we can hopefully start to make sense of what the process looked like.  I have built the core dump reporting infrastructure such that when the coredump is collected, some `gdb` commands can automatically be run, with the output dumped to the console.  Right now, the only thing it does is [print out backtraces for all threads](https://github.com/JuliaCI/julia-buildkite/blob/3ec7e645526a4b60698f38a4d0a4c828b93875fb/pipelines/main/platforms/test_linux.yml#L14-L15).  But we could conceivably add other pieces of information (such as `info files`) to dump memory mappings and whatnot.

**Pull request closed without merging**